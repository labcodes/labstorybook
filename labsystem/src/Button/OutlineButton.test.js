import React from "react";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

import OutlineButton from "./OutlineButton";

describe("OutlineButton", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<OutlineButton text="Test Outline Button" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders as expected when passing disabled as true", async () => {
    const renderedComponent = renderer
      .create(<OutlineButton text="Test Outline Disabled Button 1" disabled />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const mountedComponent = mount(
      <OutlineButton text="Test Outline Disabled Button 2" disabled />
    );
    expect(mountedComponent.find("button").prop("disabled")).toEqual(true);
  });

  it("renders as expected when passing a light skin", async () => {
    const renderedComponent = renderer
      .create(<OutlineButton text="Test Outline Light Button" skin="light" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders as expected when passing a large size", async () => {
    const renderedComponent = renderer
      .create(<OutlineButton text="Test Outline Large Button" size="large" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders as expected when passing a magnifying-glass icon", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineButton
          text="Test Outline magnifying-glass Button"
          icon="magnifying-glass"
        />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("calls props.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowedButton = shallow(
      <OutlineButton text="Test Click on Outl Button" onClick={mockOnClick} />
    );
    shallowedButton.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });

  it("renders as expected if full width", async () => {
    const renderedComponent = renderer
      .create(<OutlineButton text="Test Outline fullWidth Button" fullWidth />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
