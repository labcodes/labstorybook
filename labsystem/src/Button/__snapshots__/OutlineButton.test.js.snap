// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton renders as expected when passing a large size 1`] = `
<button
  className="btn btn--outline btn--undefined btn--large"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  
  Test Outline Large Button
</button>
`;

exports[`OutlineButton renders as expected when passing a light skin 1`] = `
<button
  className="btn btn--outline btn--light btn--normal"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  
  Test Outline Light Button
</button>
`;

exports[`OutlineButton renders as expected when passing a lupe icon 1`] = `
<button
  className="btn btn--outline btn--undefined btn--normal"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-icon lab-icon--lupe lab-icon--white lab-icon--petit lab-icon--btn__icon"
  />
  Test Outline Lupe Button
</button>
`;

exports[`OutlineButton renders as expected when passing disabled as true 1`] = `
<button
  className="btn btn--outline btn--undefined btn--normal"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  
  Test Outline Disabled Button 1
</button>
`;

exports[`OutlineButton renders with base props 1`] = `
<button
  className="btn btn--outline btn--undefined btn--normal"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  
  Test Outline Button
</button>
`;
